frontend/src/app/upload.service.ts

```diff
@@ -49,11 +54,30 @@ export class UploadService {
   }
 
   /**
-   * Updates the message field with new data. Any components that are subscribed
-   * to the message will receive the new message.
-   * @param message The data to be shared with any listening components.
+   * Updates the message field with new samples. Any components that are subscribed
+   * to the message will receive the new samples.
+   * @param samples The samples to be shared with any listening components.
    */
-  public nextMessage(message: any){
-    this.message.next(message)
+  public nextMessage(samples: any){
+    this.message.next(samples)
+  }
+
+  /**
+   * Called by side-menu.component inside sendFile(file) when a server response is received.
+   * Creates a dataset component and inserts it into the page at ref.
```

Comment:
Flip lines 66 & 67. It should start with the description of the function. Any specifics about calling order should follow after...